flocker/node/test/test_loop.py

```diff
@@ -434,6 +466,8 @@ def test_convergence_stop(self):
         # would make deployer.discover_local_state() to fail since it was
         # only configured with one result.
         action.result.callback(None)
+        reactor.advance(1.0)
```

Comment:
I think sometimes that's a reasonable strategy.  The troublesome part comes when it's legitimate for **some** stuff to be scheduled to happen, just not the one thing you don't want to happen.  It's hard/gross/fragile to sort out and ignore other unrelated delayed calls so you can assert just that the one you don't want to exist doesn't exist.

In this case it may be okay because it's really a unit test (so only a narrow bit of code gets run) and because there's no other use of `IReactorTime` anywhere near this code.  So, for right now, there's no sorting to do: the correct assertion to make is just that there are no delayed calls at all.

And actually it turns out that I think we want to do what I already made this test do **and** make that additional assertion.  Reading the test again now, I see that I asserted that we get to stopped but I didn't assert that nothing additional happens afterwards.  FWIW, the `advance(1.0)` is necessary even to get to the stopped state because I put the delay into a funny place in the chain of events.  Unfortunately it's much less convenient to put it somewhere where it doesn't get in the way of transitioning to the `STOPPED` state.  I think this is okay because transitioning to `STOPPED` is infrequent, you're not in a rush to do much of anything once you get there, and because we'll get rid of this hack at some point.